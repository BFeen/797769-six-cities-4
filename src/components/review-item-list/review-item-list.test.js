import React from "react";
import renderer from "react-test-renderer";
import ReviewItemList from "./review-item-list.jsx";


const reviewsMock = [
  {
    id: 0,
    comment: `A quiet cozy and picturesque that...`,
    dateTime: new Date(`2014-03-31T20:00:00.000Z`),
    rating: 4,
    user: {
      id: 0,
      name: `Max`,
      avatar: `img/avatar-max.jpg`,
      isPro: true,
    }
  }, {
    id: 1,
    comment: `Best place of the world!`,
    dateTime: new Date(`December 2012`),
    rating: 5,
    user: {
      id: 1,
      name: `John`,
      avatar: `https://api.adorable.io/avatars/128/5`,
      isPro: true,
    }
  }, {
    id: 2,
    comment: `The worst place of the world!`,
    dateTime: new Date(`June 2020`),
    rating: 1,
    user: {
      id: 2,
      name: `Mr. X`,
      avatar: `https://api.adorable.io/avatars/128/6`,
      isPro: false,
    }
  }, {
    id: 3,
    comment: `Nice. But too many roaches.`,
    dateTime: new Date(`May 3001`),
    rating: 3,
    user: {
      id: 3,
      name: `Leela Turanga`,
      avatar: `https://api.adorable.io/avatars/128/7`,
      isPro: false,
    }
  },
];

describe(`ReviewItemList snapshot testing`, () => {
  it(`ReviewItemList rendering`, () => {
    const tree = renderer.create(
        <ReviewItemList
          offerId={2}
          reviews={reviewsMock}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
