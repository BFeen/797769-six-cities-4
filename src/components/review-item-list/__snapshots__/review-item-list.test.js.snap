// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewItemList snapshot testing ReviewItemList rendering 1`] = `
Array [
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      4
    </span>
  </h2>,
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Photo Max"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          A quiet cozy and picturesque that...
        </p>
        <time
          className="reviews__time"
          dateTime="April 2014"
        >
          1 April
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Photo John"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://api.adorable.io/avatars/128/5"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          John
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Best place of the world!
        </p>
        <time
          className="reviews__time"
          dateTime="December 2"
        >
          1 December
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Photo Mr. X"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://api.adorable.io/avatars/128/6"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Mr. X
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          The worst place of the world!
        </p>
        <time
          className="reviews__time"
          dateTime="June 2020"
        >
          1 June
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Photo Leela Turanga"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://api.adorable.io/avatars/128/7"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Leela Turanga
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Nice. But too many roaches.
        </p>
        <time
          className="reviews__time"
          dateTime="May 3001"
        >
          1 May
        </time>
      </div>
    </li>
  </ul>,
]
`;
